content/60_gitops_for_socks_shop/20_create_a_namespace.md

```diff
@@ -0,0 +1,67 @@
++++
+title = "Creating a Namespace"
+chapter = false
+weight = 20
++++
+
+Next, declare a [namespace object](https://kubernetes.io/docs/concepts/overview/working-with-objects/namespaces/).
+This will allow scoping the deployment of the Sock Shop's components to that namespace.
+
+To do this, we need to create a stack reference to the project where we created the eks infrastructure so we can get the
+Kubeconfig and be able to build the correct kubernetes provider. We are going to make the stack reference name configurable:
+
+```typescript
+import * as pulumi from "@pulumi/pulumi";
+
+const pulumiConfig = new pulumi.Config();
+
+// Existing Pulumi stack reference in the format:
+// <organization>/<project>/<stack> e.g. "myUser/myProject/dev"
+const clusterStackRef = new pulumi.StackReference(pulumiConfig.require("clusterStackRef"));
+const argoStackRef = new pulumi.StackReference(pulumiConfig.require("argoCDStackRef"))
+```
+
+Now we can get the kubeconfig from the eks cluster for use in our provider. Append this to your `index.ts` file:
+
+```typescript
+// Get the kubeconfig from the cluster stack output.
+const kubeconfig = clusterStackRef.getOutput("kubeconfig");
+
+// Create the k8s provider with the kubeconfig.
+const provider = new k8s.Provider("k8sProvider", { kubeconfig });
+```
+
+Now we can create the namespace using the provider we just created. Append this to your `index.ts` file:
+
+```typescript
+let name = "sock-shop"
```

Comment:
const vs let

Reply:
Nah - on preview and apply it was generating perpetual diffs as it was re-encrypting each time